src/crypto/util.ts

```diff
@@ -27,8 +27,13 @@ export function legacyDeriveChildKey(
         password.byteOffset + seed.length,
         8
     );
-    view.setInt32(0, index);
-    view.setInt32(4, index >= 0 ? 0 : 0xFF);
+    if (index === 0xFFFFFFFFFF) {
```

Comment:
After the discussion we had on Slack, can you update this to
1) Hardcode `0xFF_FFFFFFFF` when the `index` is exactly that instead of using `index >>> 32`
2) Add a if branch where if `index > 0xFF_FFFFFFFF` we throw an error saying we don't support indexes of that size.